Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -676,35 +676,24 @@ static void registerLogHook()
         if (msg->buffer_sz > 1024)
             return;
 
+        // Skip faults and debug logs in non-internal builds.
+        if (type == OS_LOG_TYPE_FAULT || type == OS_LOG_TYPE_DEBUG)
+            return;
+
         CString logFormat(msg->format);
         CString logChannel(msg->subsystem);
         CString logCategory(msg->category);
 
-        Vector<uint8_t> buffer(msg->buffer, msg->buffer_sz);
-        Vector<uint8_t> privdata(msg->privdata, msg->privdata_sz);
-
-        static NeverDestroyed<Ref<WorkQueue>> queue(WorkQueue::create("Log Queue", WorkQueue::QOS::Background));
-
-        queue.get()->dispatch([logFormat = WTFMove(logFormat), logChannel = WTFMove(logChannel), logCategory = WTFMove(logCategory), type = type, buffer = WTFMove(buffer), privdata = WTFMove(privdata)] {
-            os_log_message_s msg = { 0 };
-
-            msg.format = logFormat.data();
-            msg.buffer = buffer.data();
-            msg.buffer_sz = buffer.size();
-            msg.privdata = privdata.data();
-            msg.privdata_sz = privdata.size();
-
-            char* messageString = os_log_copy_message_string(&msg);
-            if (!messageString)
-                return;
-            IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
+        char* messageString = os_log_copy_message_string(&msg);
+        if (!messageString)
+            return;
+        IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
 
-            auto connectionID = WebProcess::singleton().networkProcessConnectionID();
-            if (connectionID)
-                IPC::Connection::send(connectionID, Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(logChannel.bytesInludingNullTerminator(), logCategory.bytesInludingNullTerminator(), logString, type, getpid()), 0);
+        auto connectionID = WebProcess::singleton().networkProcessConnectionID();
+        if (connectionID)
+            IPC::Connection::send(connectionID, Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(logChannel.bytesInludingNullTerminator(), logCategory.bytesInludingNullTerminator(), logString, type, getpid()), 0);
 
-            free(messageString);
-        });
+        free(messageString);
```

Comment:
If you do the `std::unique_ptr` thing above, you don't need to have the explicit `free` here.

Reply:
> If you do the `std::unique_ptr` thing above, you don't need to have the explicit `free` here.

Fixed!

Thanks for reviewing!